compiler/passes/src/code_generation/visit_program.rs

```diff
@@ -167,7 +166,7 @@ impl<'a> CodeGenerator<'a> {
         output_string
     }
 
-    fn visit_function(&mut self, function: &'a Function) -> String {
+    fn visit_function_with(&mut self, function: &'a Function, mut futures: &[Location]) -> String {
```

Comment:
Is it cleaner to not make this mutable, and instead zip it with `function.input` in the for loop?

Reply:
Ah good point!